### PR TITLE
feat(routing): respect pageExtensions for app/pages route discovery

### PR DESCRIPTION
**Pull request by illegalcall**
feat(routing): respect pageExtensions for app/pages route discovery

- read next.config pageExtensions into resolved config
- add shared file matcher for extension-aware scanning
- apply matcher to App Router and Pages/API route scanners
- wire pageExtensions through plugin route generation and file watching
- add unit tests for matcher behavior and mdx positive/negative routing cases

Fixed: https://github.com/cloudflare/vinext/issues/251

### CODE REVIEW

File: packages/vinext/src/config/next-config.ts [hunk 1]
```diff
@@ -246,6 +250,7 @@ export async function resolveNextConfig(
       basePath: "",
       trailingSlash: false,
       output: "",
+      pageExtensions: normalizePageExtensions(),
```
Comment (james-elicx) [inline packages/vinext/src/config/next-config.ts]:
```suggestion
      pageExtensions: [],
```
nit

File: packages/vinext/src/index.ts [hunk 2]
```diff
@@ -597,8 +598,9 @@ export default function vinext(options: VinextOptions = {}): Plugin[] {
    * This is the entry point for `vite build --ssr`.
    */
   async function generateServerEntry(): Promise<string> {
-    const pageRoutes = await pagesRouter(pagesDir);
-    const apiRoutes = await apiRouter(pagesDir);
+    const pageRoutes = await pagesRouter(pagesDir, nextConfig?.pageExtensions);
+    const apiRoutes = await apiRouter(pagesDir, nextConfig?.pageExtensions);
+    const fileMatcher = createValidFileMatcher(nextConfig?.pageExtensions);
```
Comment (james-elicx) [inline packages/vinext/src/index.ts]: we could potentially create one matcher and re-use instead of creating one in each router and then here

Comment (illegalcall) [reply packages/vinext/src/index.ts]: Done.
